src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -30,6 +30,9 @@ static constexpr uint16_t PATH_MAX_LEN = 4096;
 namespace simeng {
 namespace OS {
 
+// Forward declare SimOS so as to resolve the circular dependency.
```

Comment:
This is not worded well, could change to something like "Forward declare SimOS class to resolve the circular dependency."

Reply:
done